craft_parts/filesystem_mounts.py

```diff
@@ -0,0 +1,189 @@
+
+        :return: The newly created dictionary.
+
+        """
+        return self.model_dump(by_alias=True)
+
+
+class FilesystemMount(RootModel):
+    """FilesystemMount defines the order in which devices should be mounted."""
+
+    root: Annotated[UniqueList[FilesystemMountItem], Field(min_length=1)]
+
+    def __iter__(self) -> Iterator[FilesystemMountItem]:  # type: ignore[override]
+        return iter(self.root)
+
+    @field_validator("root", mode="after")
+    @classmethod
+    def first_maps_to_slash(
+        cls, value: list[FilesystemMountItem]
+    ) -> list[FilesystemMountItem]:
+        """Make sure the first item in the list maps the '/' mount."""
+        if value[0].mount != "/":
+            raise ValueError("The first entry in a filesystem must map the '/' mount.")
+        return value
+
+    @classmethod
+    def unmarshal(cls, data: list[dict[str, Any]]) -> "FilesystemMount":
+        """Create and populate a new ``FilesystemMount`` object from list.
+
+        The unmarshal method validates entries in the input list, populating
+        the corresponding fields in the data object.
+
+        :param data: The list to unmarshal.
+
+        :return: The newly created object.
+
+        :raise TypeError: If data is not a list.
+        :raise pydantic.ValidationError: If the data fails validation.
+        """
+        if not isinstance(data, list):
+            raise TypeError("filesystem entry is not a list")
+
+        return cls.model_validate(
+            [FilesystemMountItem.unmarshal(item) for item in data]
+        )
+
+    def marshal(self) -> list[dict[str, Any]]:
+        """Create a list containing the filesystem_mount data.
+
+        :return: The newly created list.
+
+        """
+        return cast(list[dict[str, Any]], self.model_dump(by_alias=True))
+
+
+FilesystemMounts = SingleEntryDict[Literal["default"], FilesystemMount]
+
+
+def validate_filesystem_mount(data: list[dict[str, Any]]) -> None:
+    """Validate a filesystem_mount.
+
+    :param data: The filesystem mount data to validate.
+
+    :raises: ValueError if the filesystem mount is not valid.
+    """
+    FilesystemMount.unmarshal(data)
+
+
+def validate_filesystem_mounts(filesystem_mounts: dict[str, Any] | None) -> None:
+    """Validate the filesystems section.
+
+    If filesystems are defined then both partition and overlay features must
+    be enabled.
+    A filesystem_mounts dict must only have a single "default" entry.
+    The first entry in default must map the '/' mount.
+
+    :raises: FilesystemMountError if the filesystem mounts are not valid.
+    """
+    if not filesystem_mounts:
+        return
+
+    if (
+        not features.Features().enable_partitions
+        or not features.Features().enable_overlay
+    ):
+        raise errors.FilesystemMountError(
+            brief="Missing features to use filesystems",
+            details="Filesystems are defined but partition feature or overlay feature are not enabled.",
```

Comment:
I notice that the other errors offer resolutions. Is there a reason this one does not?

```suggestion
            resolution="Enable both the partition and overlay features.",
```